### PR TITLE
*: support concurrent auto analyze

### LINKED ISSUE #53460: concurrency auto analyze
## Enhancement

Now, TiDB's analyze can only be executed by a single goroutine. so when we use millions of tables, it will be very slow, so we need to implement the concurrently auto analyze for raising the QPS of auto analyze.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #53460

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: pkg/statistics/handle/autoanalyze/refresher/refresher.go [hunk 1]
```diff
@@ -103,18 +106,25 @@ func (r *Refresher) PickOneTableAndAnalyzeByPriority() bool {
...
+				statslogutil.StatsLogger().Error(
+					"Execute auto analyze job failed",
+					zap.Stringer("job", job),
+					zap.Error(err),
+				)
+			}
+		})
+		cnt++
+		if cnt >= int(variable.AutoAnlayzeConcurrency.Load()) {
+			return true
```
Comment (AilinKid) [inline pkg/statistics/handle/autoanalyze/refresher/refresher.go:124]:
```suggestion
			break
```

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

File: pkg/statistics/handle/autoanalyze/refresher/refresher.go (hunk 1 above)
Comment (hawkingrei) [reply pkg/statistics/handle/autoanalyze/refresher/refresher.go]: Thanks

**Review by AilinKid — APPROVED** [review]

**Pull request merged**